bot/feedback/feedback.py

```diff
@@ -0,0 +1,303 @@
+
+from datetime import datetime
+from typing import Optional
+import discord
+from discord.ext import commands
+
+from bot import constants
+from bot.persistence import DatabaseConnector
+from bot.logger import command_log, log
+from bot.feedback import SuggestionStatus
+
+
+class FeedbackCog(commands.Cog):
+    """Cog for Feedback Functions."""
+
+    def __init__(self, bot):
+        """Initializes the Cog.
+
+        Args:
+            bot (discord.ext.commands.Bot): The bot for which this cog should be enabled.
+        """
+        self.bot = bot
+        self._db_connector = DatabaseConnector(constants.DB_FILE_PATH, constants.DB_INIT_SCRIPT)
+
+    @commands.group(name="suggestion", hidden=True, invoke_without_command=True, aliases=["suggest"])
+    @command_log
+    async def manage_suggestions(self, ctx: commands.Context, *, suggestion: str):
+        """Command Handler for the `suggestion` command.
+
+        Allows users to submit suggestions for improving our server. After submitting, an embed containing the provided
+        information will be posted in the configured suggestion channel by SAM.
+        A suggestion can be marked as "approved", "denied", "considered" or "implemented". For each of these statuses
+        exists a corresponding subcommand.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion (str): The suggestion provided by the user.
+        """
+        await ctx.message.delete()
+        suggestion_id = self._db_connector.add_suggestion(ctx.author.id, ctx.message.created_at)
+
+        embed = _build_suggestion_embed(ctx.author, suggestion, suggestion_id)
+        message = await ctx.guild.get_channel(constants.CHANNEL_ID_SUGGESTIONS).send(embed=embed)
+
+        self._db_connector.set_suggestion_message_id(suggestion_id, message.id)
+
+        await message.add_reaction(constants.EMOJI_UPVOTE)
+        await message.add_reaction(constants.EMOJI_DOWNVOTE)
+
+    @manage_suggestions.command(name="approve")
+    @commands.is_owner()
+    @command_log
+    async def suggestion_approve(self, ctx: commands.Context, suggestion_id: int, *, reason: Optional[str]):
+        """Command Handler for the `suggestion` subcommand `approve`.
+
+        Allows the owner to mark a suggestion as approved and add an optional reason to it. If this is the case, the
+        corresponding embed will be adapted to represent this change and the user who submitted the idea will be
+        notified via DM.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion_id (int): The id of the suggestion which should be approved.
+            reason (Optional[str]): The reason for this decision.
+        """
+        await ctx.message.delete()
+        await self._change_suggestion_status(SuggestionStatus.APPROVED, ctx.author, ctx.guild, suggestion_id, reason)
```

Comment:
I don't think the order of the arguments reflects their importance or the sense of the method. 

You could reorder it to (suggestion_id, suggestion_status, author, guild, reason). 
This would make it feel like you read: 
Change the suggestion Status of <id> to <suggestionstatusvalue> using/with help of author, guild and reason.